include/ck_tile/ops/gemm/pipeline/tile_gemm_traits.hpp

```diff
@@ -3,23 +3,124 @@
 
 #pragma once
 
+#include "ck_tile/core.hpp"
+
 namespace ck_tile {
 
-template <bool kPadA_,
-          bool kPadB_,
-          bool kPadC_,
+template <bool kPadM_,
+          bool kPadN_,
+          bool kPadK_,
+          typename BlockGemmShape_,
+          typename ADataType_,
+          typename BDataType_,
+          typename CDataType_,
           typename ALayout_,
           typename BLayout_,
           typename CLayout_>
 struct TileGemmTraits
 {
-    static constexpr bool kPadA = kPadA_;
-    static constexpr bool kPadB = kPadB_;
-    static constexpr bool kPadC = kPadC_;
+    using BlockGemmShape = remove_cvref_t<BlockGemmShape_>;
```

Comment:
In FMHA design: a problem is used store data type & shape informations; and a traits is used to store kernel implementation details, for example: vector size, block size, padding or not. 
So I think you need to extract impl irrelevant parameters out and put them into `UniversalGemmPipelineProblem<>` & `GemmPipelineProblem<>`

Reply:
Sure, Thank you for the comment! I will do that!